phoenix-core/src/it/java/org/apache/phoenix/end2end/BackwardCompatibilityIT.java

```diff
@@ -319,7 +323,26 @@ public void testUpdatedSplitPolicyForSysTask() throws Exception {
                 + compatibleClientVersion,
```

Comment:
We should rename this test since we are no longer just checking that the split policy is updated. Also, better if we can do something like the following:
1. Connect with an old client: If <4.15, SYS.TASK doesn't even exist, so nothing to do. If 4.15, confirm that SYS.TASK **doesn't** have the coproc loaded. 
2. Then connect with a new client (4.16) which will trigger the metadata upgrade and update the coproc. Now do your check for existence of the coproc. Similarly can be done for the split policy too.

Reply:
Done. Thanks @ChinmaySKulkarni